CONTRIBUTING.md

```diff
@@ -323,8 +323,8 @@ Before you can start, you need a GitHub account, here are a few suggestions:
     change the code by default.
 
 - Code specifics
-  - All significant new classes, methods and functions have Doxygen-style
-    documentation in source comments.
+  - All new classes, methods and functions have Doxygen-style documentation in
+    source comments.
```

Comment:
Maybe only require Doxygen comments for public APIs? Comments for private member functions can be very useful, but I'm not sure we need Doxygen docs for them.

Reply:
This is a good point. My concern is that even internal methods and member data be documented for our own benefit. But these don't need Doxygen comments.

Reply:
I would argue that requiring doxygen even for internal members is a good idea. Yes some of it will be boilerplate, but it forces people to be explicit about what exactly input parameters and return objects are, even if it seems "obvious" to the author. It also helps with tooltips when using IDEs.

Reply:
It also avoids the question "what is a public API". I have had to inherit from MFEM data structures in the past and the undocumented private methods and data meant I had to reverse engineer a lot of MFEM internals.

Reply:
> "what is a public API"

This has come-up in the past, and I thought we had a clear answer, but maybe not. To me, the public API is the one that is exposed in the MFEM Examples and Miniapps.

Reply:
> To me, the public API is the one that is exposed in the MFEM Examples and Miniapps
I have personally experience a jump in difficult going from learning how to use some functionality shown in the examples to learning how to generalize that functionality by using public methods not in the examples.  In that sense, I personally would find it helpful for doxygen style comments on all public methods.  Maybe the verbosity of the doxygen comments can be required to be higher for methods considered part of the API versus "just public".